src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/CSharpFormattingPassBase.cs

```diff
@@ -487,12 +487,16 @@ private static SyntaxNode FixOwnerToWorkaroundCompilerQuirks(SyntaxNode owner)
         // Workaround for https://github.com/dotnet/aspnetcore/issues/36689
         // A tags owner comes back as itself if it is preceeded by a HTML comment,
         // because the whitespace between the comment and the tag is reported as not editable
-        if (owner is MarkupTextLiteralSyntax &&
-            owner.PreviousSpan() is MarkupTextLiteralSyntax literal &&
-            literal.ContainsOnlyWhitespace() &&
-            literal.PreviousSpan()?.Parent is MarkupCommentBlockSyntax)
+
+        // Get to the outermost node first. eg in "<span" we might be on the node for the "<", which is parented
+        // by some other intermediate node, which is parented by the actual start tag node. We need to get out to
+        // the start tag node, in order to reason about its siblings. The siblings of the "<" are not helpful :)
+        var outerNode = owner.GetOutermostNode();
+        if (outerNode is not null &&
+            outerNode.TryGetPreviousSibling(out var whiteSpace) && whiteSpace.ContainsOnlyWhitespace() &&
+            whiteSpace.TryGetPreviousSibling(out var comment) && comment is MarkupCommentBlockSyntax)
```

Comment:
Is this accurate? Previously, it checked `literal.PreviousSpan()?.Parent` but the parent isn't checked anymore.

Reply:
Yep. `PreviousSpan` would return the close angle of the comment, so needed to get to the parent to check if it was a comment. In this case `TryGetPreviousSibling` is not only faster, it more closely matches what the code wants to do anyway.

The scenario is covered by a couple of tests here: https://github.com/dotnet/razor/blob/main/src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Formatting/HtmlFormattingTest.cs#L1020